drivers/SmartThings/matter-switch/src/aqara-cube/init.lua

```diff
@@ -126,53 +123,66 @@ local function configure_buttons(device)
   end
 end
 
-local function device_added(driver, device)
-  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
-    local MS = get_reordered_endpoints(driver, device)
-    local main_endpoint = device.MATTER_DEFAULT_ENDPOINT
-    if #MS > 0 then
-      main_endpoint = MS[1] -- the endpoint matching to the non-child device
-      if MS[1] == 0 then main_endpoint = MS[2] end -- we shouldn't hit this, but just in case
-    end
-    device.log.debug("main button endpoint is "..main_endpoint)
-
-    -- At the moment, we're taking it for granted that all momentary switches only have 2 positions
-    local current_component_number = 1
-    local component_map = {}
-    for _, ep in ipairs(MS) do -- for each momentary switch endpoint (including main)
-      device.log.debug("Configuring endpoint "..ep)
-      -- build the mapping of endpoints to components
-      component_map[string.format("%d", current_component_number)] = ep
-      current_component_number = current_component_number + 1
+local function set_configure(driver, device)
```

Comment:
Hi, I was wondering which part of `device_added` is specifically needed to run on each init (is it the endpoint configuration or the profile update for example). Could we put those parts in `set_configure` and leave the rest in `device_added`? Or is everything needed to run on each init?

Reply:
@nickolas-deboom This Aqara Cube update requires both endpoint configuration and the profile update. In particular, when the driver is updated, only the `init` is called, so I think both of the above should be moved to the `init`.